src/client/dfs/dfs_internal.c

```diff
@@ -0,0 +1,31 @@
+/**
+ * (C) Copyright 2018-2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include <daos/common.h>
+#include "dfs_internal.h"
+
+void
```

Comment:
why is this in dfs_internal.c while the other is in dfs.c?

Reply:
The “daos/common.h” header provides the declaration of D_FREE. Usually, I try to limit the exposure of the header files. Since D_FREE() is only used on this file. I decided to include its header here.

Reply:
*scope

Reply:
this is probably not even necessary, since we can just call free on the array?

Reply:
If I understood correctly Jeff comments. Our memory leak detection infrastructure relies on the calls to D_ALLOC* and D_FREE. So, we have to explicitly call D_FREE. So, that’s why we cannot call libc.so::free() directly from the Python’s code.

Reply:
Yes, potential log parsing based error detection is the issue with just calling free.